### ISSUE TITLE
SSE writes are ignored by timeout settings

### ISSUE THREAD

**Issue by andrewbanchich** [ory/oathkeeper#877]
### Preflight checklist

- [X] I could not find a solution in the existing issues, docs, nor discussions.
- [X] I agree to follow this project's [Code of Conduct](https://github.com/ory/oathkeeper/blob/master/CODE_OF_CONDUCT.md).
- [X] I have read and am following this repository's [Contribution Guidelines](https://github.com/ory/oathkeeper/blob/master/CONTRIBUTING.md).
- [ ] This issue affects my [Ory Cloud](https://www.ory.sh/) project.
- [X] I have joined the [Ory Community Slack](https://slack.ory.sh).
- [X] I am signed up to the [Ory Security Patch Newsletter](https://ory.us10.list-manage.com/subscribe?u=ffb1a878e4ec6c0ed312a3480&id=f605a41b53).

### Describe the bug

`serve.proxy.timeout.write` will close server-sent event connections, even if they are using keep-alives.

### Reproducing the bug

1. Create server with SSE endpoint and keep-alives sent every 15 seconds.
2. On the frontend, use the JS `EventSource` API to connect and process messages.
3. The browser console should display an error after the Oathkeeper's timeout setting (120s by default) saying the connection was closed.

### Version

0.38.15-beta.1

### On which operating system are you observing this issue?

Linux

### In which environment are you deploying?

Docker

**Comment by aeneasr** [comment]
Hm yeah I think Oathkeeper really struggles with SSE. I think one option is to increase the timeout values. However, it will still close at some point.

Maybe we could implement keep alive somehow?

**Comment by andrewbanchich** [comment]
Yeah, having keep-alive would be ideal and allow Oathkeeper to differentiate between a working SSE connection and one that needs to be closed.

**Comment by andlinger** [comment]
Maybe it is possible to simply suspend the timeout if the content type is "text/event-stream".
